### PR TITLE
WIP: Add name color to color picker

### PR DESCRIPTION
**Pull request by nagauta**
## Checklist

- [ ] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ ] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ ] I checked that files in the `assets` folder are used by the extension itself
- [ ] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/color-picker/package.json [hunk 1]
```diff
@@ -76,6 +76,12 @@
       "name": "convert-color",
       "title": "Convert Color",
       "description": "Convert any color to a different format.",
+      "mode": "no-view"
+    },
+    {
+      "name": "name-color",
+      "title": "Name Color",
+      "description": "name the color selected.",
```
Comment (thomaspaulmann) [inline extensions/color-picker/package.json:84]: I would suggest a similar naming than the `Pick Color` command:
```suggestion
      "description": "Name a color on your desktop",
```

**Comment by LitoMore** [comment]
I've noticed most of the code from `name-color` is the same as the `pick-color` command.

You may be able to re-organize the reused part to polish the code.

File: extensions/color-picker/src/name-color.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    if (props.launchContext?.callbackLaunchOptions) {
+      if (props.launchContext?.copyToClipboard) {
+        await Clipboard.copy(colorName);
+      }
+
+      try {
+        await callbackLaunchCommand(props.launchContext.callbackLaunchOptions, { hex });
+      } catch (e) {
+        await showFailureToast(e);
+      }
```
Comment (LitoMore) [inline extensions/color-picker/src/name-color.ts:40]: We can callback with both `hex` and `colorName` here. The API documentation needs to be updated as well.

```suggestion
      try {
        await callbackLaunchCommand(props.launchContext.callbackLaunchOptions, { hex, colorName });
      } catch (e) {
        await showFailureToast(e);
      }
```

We can also add the `colorName` callback payload to `pick-color` command.

I can help if you need any help on composing documentation and `callLaunchCommand` functions.

File: extensions/color-picker/package.json (hunk 1 above)
Comment (nagauta) [reply extensions/color-picker/package.json:84]: fixed, thanks!

File: extensions/color-picker/src/name-color.ts (hunk 2 above)
Comment (LitoMore) [reply extensions/color-picker/src/name-color.ts]: We should make the API convenient. So let's return both `hex` and `colorName`.

**Comment by LitoMore** [comment]
We can make this feature available in UI. So that users can pick their favorite one by platform type:

- roygbiv
- HTML
- x11
- pantone
- ntc

The UX workflow can be:

1. Run `name color` command, let user to pick a color
2. After picking a color, return the UI with the name list

It also facilitates the addition of AI functions. So that I can simply add a new action to provide the Raycast Pro enhanced AI feature.

For this implementation, support for [Corss-Extension](https://github.com/LitoMore/cross-extension-conventions) is no longer needed. We can implement the name color API to `pick-color` command only.

**Comment by LitoMore** [comment]
@nagauta Are you still working on this?

**Comment by nagauta** [comment]
> @nagauta Are you still working on this?

Yes
I have to take google cloud exam today 
It's finished and passed.
I'm on doing this again

File: extensions/color-picker/src/name-color.ts (hunk 2 above)
Comment (nagauta) [reply extensions/color-picker/src/name-color.ts]: fine, I mod as you advice

**Pull request closed without merging**